### ISSUE TITLE
Proper redirect using Credentials strategy

### ISSUE THREAD

**Issue by emdede** [OrJDev/solid-auth#3]
I have two problems with current behavior. Only one of which would need to be solved for me personally.

1. Using Credentials strategy generates strange-looking URLs to redirect to on failed login attempt. This is the URL generated in my app for example: `http://localhost:3000/login?callbackUrl=http%3A%2F%2Flocalhost%3A3000%2Flogin&error=CredentialsSignin`. For context, I am using `pages` option because I don't want to use the built-in login form. It doesn't actually redirect me, which is good, but the URL should remain the same.

2. Using `redirect: false` and navigating manually will flash the Login page after Session has initialized. The session now being a resource triggers my global Suspense boundary while trying to log in. That's fine, but after the session has loaded and the Suspense goes away, it shows my Login page one more time. I am using a `createRenderEffect` to navigate as soon as possible but it's still not early enough to prevent a flashing of the Login page. `Login -> Suspense -> Login -> Navigate`

I have tried to come up with solutions on my end but I think I can't make it work reliably without changing `solid-auth` source. Do you think it would be possible to "fix" either one of these? I am guessing the first one would be much easier to implement. I have played around with it a bit but I don't feel confident in making changes because I don't know how other Providers might be affected.

Cheers!

**Comment by OrJDev** [comment]
> I have two problems with current behavior. Only one of which would need to be solved for me personally.
> 
> 
> 
> 1. Using Credentials strategy generates strange-looking URLs to redirect to on failed login attempt. This is the URL generated in my app for example: `http://localhost:3000/login?callbackUrl=http%3A%2F%2Flocalhost%3A3000%2Flogin&error=CredentialsSignin`. For context, I am using `pages` option because I don't want to use the built-in login form. It doesn't actually redirect me, which is good, but the URL should remain the same.
> 
> 
> 
> 2. Using `redirect: false` and navigating manually will flash the Login page after Session has initialized. The session now being a resource triggers my global Suspense boundary while trying to log in. That's fine, but after the session has loaded and the Suspense goes away, it shows my Login page one more time. I am using a `createRenderEffect` to navigate as soon as possible but it's still not early enough to prevent a flashing of the Login page. `Login -> Suspense -> Login -> Navigate`
> 
> 
> 
> I have tried to come up with solutions on my end but I think I can't make it work reliably without changing `solid-auth` source. Do you think it would be possible to "fix" either one of these? I am guessing the first one would be much easier to implement. I have played around with it a bit but I don't feel confident in making changes because I don't know how other Providers might be affected.
> 
> 
> 
> Cheers!

Pages can be modified from the authOptions, so error & login can be "/", and regarding the redirect: false, it shouldn't flash unless you have a lack of suspense boundaries, so make sure you actually wrap this entire thing with a suspense, also how does this look like (code wise) on your end?

**Comment by emdede** [comment]
> Pages can be modified from the authOptions, so error & login can be "/", and regarding the redirect: false, it shouldn't flash unless you have a lack of suspense boundaries, so make sure you actually wrap this entire thing with a suspense, also how does this look like (code wise) on your end?

I have just submitted a simplified version of my code as a [docs PR](https://github.com/OrJDev/solid-auth/pull/4). 

You are right. Just adding an additional `<Suspense>` to my Login route did the trick. I assumed it wouldn't make a difference being wrapped under the Global Suspense. Thanks. With this, my problem is solved. :)

Edit: for reference, I navigate using this:
```tsx
  const session = createSession()
  createRenderEffect(() => {
    session()?.user && navigate('/')
  })
  ```
This is meant to also redirect when a User is signed in already and visits login page.

**Comment by OrJDev** [comment]
Awesome glad it is working 🤶,
Also if you create a new app using create jd app it will already wrap your `createSession` calls with suspense

**Comment by emdede** [comment]
I'm sorry for the confusion. This did not actually work. I tried several different ways with `redirect: false` and default options. I accidentally did it with the wrong one which made me think it was solved, but the problem remains. With `redirect` enabled, I get instant redirect on success but weird `window.location.href` value on failure. With `redirect: false` I still get a flashing Login page on success. :/

**Comment by OrJDev** [comment]
> I'm sorry for the confusion. This did not actually work. I tried several different ways with `redirect: false` and default options. I accidentally did it with the wrong one which made me think it was solved, but the problem remains. With `redirect` enabled, I get instant redirect on success but weird `window.location.href` value on failure. With `redirect: false` I still get a flashing Login page on success. :/

So can you make a really minimized example of your sign in & sign out component

**Comment by emdede** [comment]
I created a simple example on [stackblitz](https://stackblitz.com/edit/solid-start-bare-ts-mep9zt?file=src/routes/[...404].tsx).

However, it does not work on Stackblitz. I'm assuming they have some protection which blocks requests. Anyhow, I've tested the repo locally and it builds and runs fine. Download project, `npm i`, `npm run dev` and it should work. Keep input field empty and it will error, any name will succeed. Both `signIn` options are present on login page. One is commented out.

While building this I noticed a bug(?) if you use `redirect: false` the `session` resource is triggered even though the Login attempt fails.

Edit: I cannot reliably reproduce the issues. There must be some other factors playing into it. Sometimes it works fine, sometimes it doesn't. Sometimes `logout` will route me back, sometimes it will stay on the page.

Edit2: Have to set low network speed for flashing to occur. But the `signOut` redirect is still unpredictable.

What I've noticed is that when I login via `redirectTo` I will not be navigated back to my login page after `signOut`. When logging in with `redirect: false` it works. Trying `signOut({ redirectTo: '/' })` doesn't work in both cases.
